File: week15-3.py
class Solution:#week15-3 py.
    def equalSubstring(self, s: str, t: str, maxCost: int) -> int:
        ans = 0
        N = len(s)
        j = 0
        for i in range(N):
            maxCost -= abs( ord(s[i]) - ord(t[i]) )
            while maxCost < 0:
                maxCost += abs( ord(s[j])-ord(t[j]) )
                j += 1
            ans = max(ans, i-j+1)
        return ans

File: week15-3.c
//week15-3.c
int equalSubstring(char* s, char* t, int maxCost) {
    int ans = 0, j = 0;
    int N = strlen(s);
    for(int i=0; i<N; i++){
        maxCost -= abs(s[i] - t[i]);
        while(maxCost<0){
            maxCost += abs(s[j] - t[j]);
            j += 1;
        }
        if(i-j+1>ans) ans = i-j+1;
    }
    return ans;
}
